### PR TITLE
Refractor: created new files for ensure function.

### PR DESCRIPTION
**Pull request by crombus**
gocyclo complexity less than 10
created helper functions
https://issues.redhat.com/browse/RHSTOR-1124

### CODE REVIEW

File: pkg/controller/storagecluster/ensure_Externalstorageclusterresources.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                objectKey := types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}
+                switch d.Kind {
+                case "ConfigMap":
+                        cm := &corev1.ConfigMap{
+                                ObjectMeta: objectMeta,
+                                Data:       d.Data,
+                        }
+                        found := &corev1.ConfigMap{ObjectMeta: objectMeta}
+                        err := r.client.Get(context.TODO(), objectKey, found)
+                        r.create_configmap(err, cm, reqLogger)
```
Comment (jarrpa) [inline pkg/controller/storagecluster/ensure_Externalstorageclusterresources.go:92]: Don't pass errors into helper functions! :stuck_out_tongue_closed_eyes: Does the complexity go to high if you leave the `err != nil` stanza here?

File: pkg/controller/storagecluster/ensure_Externalstorageclusterresources.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+         } else if d == cephRbdStorageClassName {
+                // Setting the PoolName for RBD StorageClass
+                sc = scs[1]
+         } else if d == cephRgwStorageClassName {
+                // Setting the Endpoint for OBC StorageClass
+                sc = scs[2]
+         }
+}
+
+func (r *ReconcileStorageCluster) create_configmap(err error, cm *corev1.ConfigMap, reqLogger logr.Logger) error {
```
Comment (jarrpa) [inline pkg/controller/storagecluster/ensure_Externalstorageclusterresources.go:128]: `createExternalStorageClusterConfigMap()`

File: pkg/controller/storagecluster/ensure_Externalstorageclusterresources.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                        }
+                } else {
+                        reqLogger.Error(err, "unable the get the configmap")
+                        return err 
+                }
+         }
+         return nil
+}
+
+func (r *ReconcileStorageCluster) create_secret(err error, sec *corev1.Secret, reqLogger logr.Logger) error {
```
Comment (jarrpa) [inline pkg/controller/storagecluster/ensure_Externalstorageclusterresources.go:145]: `createExternalStorageClusterSecret()`

File: pkg/controller/storagecluster/ensure_ceph_objectstores.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        "github.com/go-logr/logr"
+        ocsv1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1"
+        cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+        "k8s.io/apimachinery/pkg/api/errors"
+        "k8s.io/apimachinery/pkg/types"
+)
+
+// switch case for error handling
+func (r *ReconcileStorageCluster) switchcase_cephObjectStore(cephObjectStores []*cephv1.CephObjectStore, instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
```
Comment (jarrpa) [inline pkg/controller/storagecluster/ensure_ceph_objectstores.go:15]: `createCephObjectStores()`

Comment (jarrpa) [reply pkg/controller/storagecluster/ensure_ceph_objectstores.go:15]: Also move this below the next function.

File: pkg/controller/storagecluster/ensure_ceph_objectstoresusers.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+        "github.com/go-logr/logr"
+        ocsv1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1"
+        cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+        "k8s.io/apimachinery/pkg/api/errors"
+        "k8s.io/apimachinery/pkg/types"
+)
+
+// switch case for error handling
+func (r *ReconcileStorageCluster) switchcase_cephObjectStoreUsers(cephObjectStoreUsers []*cephv1.CephObjectStoreUser, instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
```
Comment (jarrpa) [inline pkg/controller/storagecluster/ensure_ceph_objectstoresusers.go:15]: `createCephObjectStoreUsers()`

Comment (jarrpa) [reply pkg/controller/storagecluster/ensure_ceph_objectstoresusers.go:15]: Also move this function below the next function.

**Comment by jarrpa** [comment]
@crombus DId you mean to close this PR...?

**Comment by crombus** [comment]
@jarrpa NO, while resolving conflicts i messed up the commits.  I have the changes with me in local branch I'll raise new PR.

**Pull request closed without merging**